packages/react-storage/src/components/StorageBrowser/Views/Controls/Navigate.tsx

```diff
@@ -9,15 +15,85 @@ interface LocationData {
   locationType: LocationType;
 }
 
+/**
+ * <NavigateItem />
+ */
+
 type RenderNavigateItem = (props: { item: LocationData }) => React.JSX.Element;
-interface NavigateItem<T extends StorageBrowserElements>
-  extends RenderNavigateItem,
-    Pick<T, 'Button' | 'ListItem'> {
+
+interface NavigateItem<
+  T extends StorageBrowserElements = StorageBrowserElements,
+> extends RenderNavigateItem {
   Separator: T['Span'];
+  Button: T['Button'];
+  ListItem: T['ListItem'];
 }
 
-export interface NavigateControl<T extends StorageBrowserElements> {
+const Item = withBaseElementProps(ListItem, {
+  className: `${BLOCK_NAME}__list__item`,
+});
+
+const NavigateButton = withBaseElementProps(Button, {
+  className: `${BLOCK_NAME}__button`,
+});
+
+const Separator = withBaseElementProps(Span, {
+  'aria-hidden': true,
+  className: `${BLOCK_NAME}__separator`,
+  children: '/',
+});
+
+const NavigateItem: NavigateItem = () => (
```

Comment:
@calebpollman Given this structure for what NavigateItem looks like, could you give an example of how you saw the `RenderNavigateItem` interface being used? I couldn't quite wrap my head around that.